lib/factorial.rb

```diff
@@ -1,6 +1,18 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n=number
+# Space complexity: O(1), because no additional space is required other than for a couple of variables
 def factorial(number)
-  raise NotImplementedError
+  total = number
+  if number == nil
+    raise ArgumentError, "number is nil"
```

Comment:
While this works for this fn because factorial is a known concept in math, I would rethink the message. For example, `number cannot be nil` or `input cannot be nil`. These messages tell the user what the input is and why an error was thrown.